src/System.Runtime/tests/System/Version.cs

```diff
@@ -3,211 +3,263 @@
     {
-        var version = new Version(major, minor, build, revision);
-        VerifyVersion(version, major, minor, build, revision);
+        yield return new object[] { new Version(1, 2), null, 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2), 0 };
+        yield return new object[] { new Version(1, 2), new Version(2, 0), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2, 1), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2, 0, 1), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0), 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 1), 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 0, 1), 1 };
+
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 4), 0 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 5), -1 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 3), 1 };
     }
 
-    private static void VerifyVersion(Version version, int major, int minor, int build, int revision)
+    [Theory, MemberData(nameof(CompareToTestData))]
+    public static void TestCompareTo(Version version1, Version obj, int expected)
     {
-        Assert.Equal(major, version.Major);
-        Assert.Equal(minor, version.Minor);
-        Assert.Equal(build, version.Build);
-        Assert.Equal(revision, version.Revision);
-        Assert.Equal((short)(revision >> 16), version.MajorRevision);
-        Assert.Equal((short)(revision & 0xFFFF), version.MinorRevision);
+        Version version2 = obj as Version;
+
+        Assert.Equal(expected, Math.Sign(version1.CompareTo(version2)));
+        if (version1 != null && version2 != null)
+        {
+            if (expected >= 0)
+            {
+                Assert.True(version1 >= version2);
+                Assert.False(version1 < version2);
+            }
+            if (expected > 0)
+            {
+                Assert.True(version1 > version2);
+                Assert.False(version1 <= version2);
+            }
+            if (expected <= 0)
+            {
+                Assert.True(version1 <= version2);
+                Assert.False(version1 > version2);
+            }
+            if (expected < 0)
+            {
+                Assert.True(version1 < version2);
+                Assert.False(version1 >= version2);
+            }
+        }
+
+        IComparable comparable = version1;
+        Assert.Equal(expected, Math.Sign(comparable.CompareTo(obj)));
     }
 
     [Fact]
-    public void CompareToObject()
+    public static void TestCompareTo_Invalid()
     {
-        var x = (IComparable)new Version(1, 2);
-        Assert.Equal(1, x.CompareTo(null));
-        Assert.Equal(0, x.CompareTo(new Version(1, 2)));
-        Assert.True(x.CompareTo(new Version(1, 3)) < 0);
```

Comment:
I'm searching for "1, 3", and I don't see it anywhere in your new version.  Where is it?  The one you linked to is ```yield return new object[] { new Version(1, 2), null, 1 };```